tea.py

```diff
@@ -0,0 +1,460 @@
+import pybamm
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+
+
+class TEA():
+    """Techno-Economic Analysis class with functions for the estimation of cell and cell-production metrics:
+
+    Cell metrics:
+     stack_energy_densities - Calculation of gravimetric and volumetric energy densities on stack level (WIP)
+     stack_breakdown - Calculation of areal mass and volume loadings (WIP)
+     plot_stack_breakdown - Plot of areal mass and volume loadings (WIP)
+     stack_power_densities - Calculation of gravimetric and volumetric power densities on stack level (to be implemented)
+     cell_energy_densities - Calculation of gravimetric and volumetric energy densities on cell level (to be implemented)
+     cell_power_densities - Calculation of gravimetric and volumetric power densities on cell level (to be implemented)
+     cell_breakdown - Calculation of total masses and volumes on cell level (to be implemented)
+     plot_cell_breakdown - Plot of total masses and volumes on cell level (to be implemented)
+
+    Process metrics:
+     component_cost_breakdown - Calculation of total component costs on cell level (to be implemented)
+     plot_component_cost_breakdown - Plot of total component costs on cell level (to be implemented)
+     End of next month:
+     (bottom-up model for production costs (+emissions,waste) with respect to the cell design)
+     (from precursor_production to electrode_production, cell_assembly, formation up to recycling)
+     """
+    
+    # nearly all used parameter are saved in the results dict
+    # maybe example results-sets for PyBaMM parameter-sets and non Li-ion chemistries can be provided?
+
+    # In case parameter are not available in a specific parameter_values set (as for non Li-ion chemistries),
+    # the values can be predefined/supplied with results
+    # In case parameter are in general not included in parameter_values, e.g. active material density or price
+    # they are supplied with inputs to the functions
+    # In case parameter are not available in parameter_values or results, defaults are used for:
+    # current collector thicknesses and densities and for electrolyte density
+    # if values are in results and in inputs or parameter_values, the values from results are not used
+
+    def __init__(self, parameter_values=None, results=None):
+        self.parameter_values = parameter_values
+        if results is not None:
```

Comment:
can do `self.results = results or {}`

Reply:
I'm actually a little confused about how results is used and why you can pass it when you create an instance of the class?